app/controllers/concerns/phone_confirmation_flow.rb

```diff
@@ -65,7 +72,26 @@ def send_confirmation_code
     # user's session. Re-sending the confirmation code doesn't generate a new one.
     self.confirmation_code = generate_confirmation_code unless confirmation_code
 
-    SmsSenderOtpJob.perform_later(confirmation_code, unconfirmed_phone)
+    if current_otp_delivery_method == :voice
```

Comment:
Hmm. Why default to SMS to handle unsupported behavior? If the user purposefully changes the parameters in the URL, they do so at their own risk. Ideally, we would refresh the page and show them the last screen they saw, which would require storing the state. Alternatively, perhaps a better approach would be to use routes and controllers for each delivery type as opposed to parameters, and if a user tries to type in a nonexistent route, it would be a legitimate 404 and we wouldn't need to write any code to handle that case specifically.

I'll work on this when I refactor.